File: 백준/Bronze/13300. 방 배정/방 배정.java
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    private static int N;
    private static int K;
    private static StringBuilder sb;
    private static int result;

    public static void main(String[] args) throws FileNotFoundException {
        Scanner sc = new Scanner(System.in);

        N = sc.nextInt();
        K = sc.nextInt();

        int[][] arr = new int[7][2];

        //입력
        for (int i = 1; i <= N; i++) {
            int sex = sc.nextInt();
            int grade = sc.nextInt();

            arr[grade][sex] += 1;
        }

        result = 0;

        //필요한 방 수 계산
        for (int i = 1; i <= 6; i++) {
            if (arr[i][0] % K > 0) {
                result += (arr[i][0] / K + 1);
            } else {
                result += (arr[i][0] / K);
            }

            if (arr[i][1] % K > 0) {
                result += (arr[i][1] / K + 1);
            } else {
                result += (arr[i][1] / K);
            }
        }

        System.out.println(result);
    }

    private static String judge(int[] A, int[] B) {
        boolean AResult = false;
        boolean BResult = false;

        if (A[4] > B[4]) {
            AResult = true;
        } else if (A[4] < B[4]) {
            BResult = true;
        } else {
            if (A[3] > B[3]) {
                AResult = true;
            } else if (A[3] < B[3]) {
                BResult = true;
            } else {
                if (A[2] > B[2]) {
                    AResult = true;
                } else if (A[2] < B[2]) {
                    BResult = true;
                } else {
                    if (A[1] > B[1]) {
                        AResult = true;
                    } else if (A[1] < B[1]) {
                        BResult = true;
                    } else {
                        AResult = true;
                        BResult = true;
                    }
                }
            }
        }

        if (AResult && BResult) {
            return "D";
        } else if (AResult && !BResult) {
            return "A";
        } else if (BResult && !AResult) {
            return "B";
        } else {
            return "D";
        }
    }
}

File: 백준/Bronze/13300. 방 배정/방 배정.py
#남학생은 남학생끼리, 여학생은 여학생끼리
#한 방에는 같은 학년의 학생들을 배정

#한 방에 배정할 수 있는 최대 인원 수 K
#필요한 방의 최소 개수



#수학여행에 참가하는 학생 수를 나타내는 정수 N
data = input().split()
N = int(data[0])

#한 방에 배정할 수 있는 최대 인원 수 K
K = int(data[1])

arr = [[0 for j in range(2)] for i in range(7)]

#다음 N 개의 각 줄에는
for i in range(N):
    #학생의 성별 S와 학년 Y
    data = input().split()
    sex = int(data[0])
    grade = int(data[1])
    arr[grade][sex] += 1

sum = 0

for i in range(1, 7):
    for j in range(0, 2):
        sum += arr[i][j] // K
        if (arr[i][j] % K > 0):
            sum += 1

print(sum)